arbor/partition_load_balance.cpp

```diff
@@ -69,16 +77,30 @@ domain_decomposition partition_load_balance(const recipe& rec,
 
     std::vector<group_description> groups;
     for (auto k: kinds) {
-        // put all cells into a single cell group on the gpu if possible
-        if (nd.num_gpus && has_gpu_backend(k)) {
-            groups.push_back({k, std::move(kind_lists[k]), backend_kind::gpu});
+        partition_hint hint;
+        if (auto opt_hint = util::value_by_key(hint_map, k)) {
+            hint = opt_hint.value();
+        }
+
+        backend_kind backend = backend_kind::multicore;
+        std::size_t group_size = hint.cpu_group_size;
+
+        if (hint.prefer_gpu && nd.num_gpus && has_gpu_backend(k)) {
```

Comment:
I know you won't like it, but `nd.num_gpus>0` is clearer.

Reply:
Will add, no worries.